tools/wheelmaker.py

```diff
@@ -510,9 +510,30 @@ def main() -> None:
             ) as description_file:
                 description = description_file.read()
 
-        metadata = None
-        with open(arguments.metadata_file, "rt", encoding="utf-8") as metadata_file:
-            metadata = metadata_file.read()
+        metadata = arguments.metadata_file.read_text(encoding="utf-8")
+
+        # Search for any `Requires-Dist` entries that refer to other files and
+        # expand them.
+        for meta_line in metadata.splitlines():
+            if not meta_line.startswith("Requires-Dist: @"):
+                continue
+            file, _, extra = meta_line[len("Requires-Dist: @") :].partition(";")
+            extra = extra.strip()
+
+            reqs = []
+            for reqs_line in Path(file).read_text(encoding="utf-8").splitlines():
+                reqs_text = reqs_line.strip()
+                if not reqs_text or reqs_text.startswith(("#", "-")):
+                    continue
+
+                # If `;` is find the extra option should be joined to the existing
+                # set of constraints via `and`.
+                split = ";"
+                if ";" in reqs_text:
+                    split = "and"
```

Comment:
just realized that there could be an issue in the general sense if we have `python_version < "3.8" or python_version > "3.10"`, but the user should be able to workaround it with using `()`.

Maybe we could use https://packaging.pypa.io/en/stable/requirements.html and check if https://packaging.pypa.io/en/stable/markers.html contains `or` because `and` should be fine. This may also validate the marker.

Let me know if this is feasible to fix/workaround or if we just have to document this limitation.

Reply:
Great thinking, Updated code and test!